### PR TITLE
Fix aten::nonzero. Add aten::scalar_tensor overload for sym number

### PR DESCRIPTION
**Pull request by BowenBao**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #931

Fix aten::nonzero impl and test.
Fix aten::scalar_tensor test.
Add aten::scalar_tensor overload for sym number.

### CODE REVIEW

**Comment by justinchuby** [comment]
Curious why we need a transpose?

**Comment by BowenBao** [comment]
> Curious why we need a transpose?

Because in torch the return shape is [n, d], while in onnx [d, n]. `d` is rank of input tensor, `n` is number of nonzero elements.

**Review by justinchuby — APPROVED** [review]

File: onnxscript/tests/function_libs/torch_lib/ops_test_data.py [hunk 1]
```diff
@@ -1007,9 +1015,15 @@ def _where_input_wrangler(
...
-        matcher=lambda sample: sample.kwargs.get("as_tuple") is not None,
+        input_wrangler=_nonzero_input_wrangler,
+    )
+    .xfail(
+        matcher=lambda sample: sample.kwargs.get("as_tuple"),
         reason="as_tuple=True is not supported",
+    )
+    .xfail(
+        matcher=lambda sample: len(sample.input.shape) == 0,
+        reason="Output 'shape' do not match: torch.Size([0, 1]) != torch.Size([0, 0]).",
```
Comment (justinchuby) [inline onnxscript/tests/function_libs/torch_lib/ops_test_data.py:1026]: Need to fixme for this one?

**Comment by justinchuby** [comment]
> > Curious why we need a transpose?
> 
> Because in torch the return shape is [n, d], while in onnx [d, n]. `d` is rank of input tensor, `n` is number of nonzero elements.

Would be nice to have this in comment. Thanks!

File: onnxscript/tests/function_libs/torch_lib/ops_test_data.py (hunk 1 above)
Comment (BowenBao) [reply onnxscript/tests/function_libs/torch_lib/ops_test_data.py]: done

**Comment by BowenBao** [comment]
@justinchuby updated PR wrt aten::tensor. PTAL thanks!

**Comment by BowenBao** [comment]
Proceed to merge, will address any more feedbacks in followup.

**Comment by justinchuby** [comment]
LGTM!

**Pull request merged**